### ISSUE TITLE
Vi Mode: 'cw' operation doesn't always work correctly

### ISSUE THREAD

**Issue by gwojan** [PowerShell/PSReadLine#218]
If I attempt to modify a line like 'vim .\PSReadLine\VisualEditing.vi.cs' there are issues with `cw` when editing between non-word characters. For example:

Move the cursor to the first instance of PSReadLine and press `cw` the results are:
`vim .\ \VisualEditing.vi.cs`. Notice there is a space between `\ \`. The same occurs trying to `cw` 'VisualEditing' or 'vi'. The operation `dw` or `dwi` works correctly.

**Comment by srdubya** [comment]
Fixed by improving the handling of non-whitespace word delimiters.

**Comment by gwojan** [comment]
Thanks Steve. I haven't found any cases where the space is being added now. However, I've found a few other cases where `cw` and even `dw` don't behave correctly.

On the string `$response.Headers['location']` move the cursor to 'l' in location press `cw`. The word 'location' is deleted along with the closing single quote and the cursor is moved outside the ']' looking something like `$response.Headers[']`.

On the string `test(int)` try to `cw` 'int'. In this case 'int' is deleted but the cursor is moved outside the closing `)`.

On the string `test()` move the cursor to `(` and type `cw`, the `(` is deleted and the cursor is moved after the `)`.

On the string `test()` move the cursor to `(` and type `dw`, the `(` is deleted and the cursor is moved under the `)`.

Basically the above behaviors also occur with `{}` and `[]` as well.

**Comment by srdubya** [comment]
Fixed the additional cases added to #218.  Thanks for trying these cases and submitting the problems you found.  It helps to have another pair of hands trying more permutations.

**Comment by gwojan** [comment]
It's my pleasure! Thanks for taking the time to keep this fork alive. With VsVim in Visual Studio and PSReadLine in PowerShell I'm just so much more productive -- muscle memory and all that. Now if there was a usable Vi mode for Word I could really be fat, dumb and happy. ;-)

**Comment by lzybkr** [comment]
Closing presumably fixed vi mode issues.
